daemon/start_unix.go

```diff
@@ -19,6 +19,9 @@ func (daemon *Daemon) getLibcontainerdCreateOptions(container *container.Contain
 	}
 
 	rt := daemon.configStore.GetRuntime(container.HostConfig.Runtime)
+	if rt == nil {
+		rt = daemon.configStore.GetRuntime(daemon.configStore.GetDefaultRuntimeName())
+	}
```

Comment:
Actually wondering if this is the correct fix; this prevents a `panic`, but would use the wrong runtime. We already set "default" to use as runtime is nothing was configured in the container (lines above; https://github.com/moby/moby/blob/d5612a0ef8aa5c43f8345302e8d20b5fd25517b3/daemon/start_unix.go#L11-L15)

If we don't get a runtime at this point, it means a runtime was configured that was not found in the deamon configuration.

I _think_ the correct thing to do in this case would be to return an error (e.g., "runtime not found"), but we'd have to check if that error needs special handling somewhere.

Having a quick look at the "restore" stage (when starting the daemon), and that looks to be "ok" (it won't prevent the daemon from starting, but will log an error; https://github.com/moby/moby/blob/64bd4485b3a66a597c02c95f5776395e540b2c7c/daemon/daemon.go#L535-L539

Reply:
**sorry for page works.** 

Error is better. 
And container status is dead when runtime is not suitable maybe better.
Then caller could not to retry invoke `start` by report create/start container process is failed.

Reply:
Discussing with @tiborvass @tianon the name of the default runtime changed from `docker-runc` to `runc` in https://github.com/moby/moby/commit/34eede0296bce6a9c335cb429f10728ae3f4252d#diff-f3d6dc48946af46cd74232848bef541845f1101b43a5d0239bc0ed9767f39a38L79-R79 (Docker 19.03 and up)

We should probably add a migration step (during container restore) or a special case for that name (and for `docker-runc` change it to the new default name.

For other "unknown" runtimes, we should return an error

Possibly we can do that on lines 16-18 above 🤔